EIPS/eip-1024.md

```diff
@@ -0,0 +1,102 @@
+---
+eip: 1024
+title: Add web3.eth.encrypt and web3.eth.decrypt functions
+author: Tope Alabi <alabitemitope@gmail.com>
+status: Draft
+type: Interface Track
+created: 2018-05-14
+---
+
+
+### Abstract
+This EIP proposes a cross-client method for requesting encryption/decryption. This method will include a version parameter, so that different encryption methods can be added under the same name. Nacl is a cryptographically complete and well audited library that works well for this by implementers are free to choose their crypto. Ethereum keypairs should not be used directly for encryption, instead we should derive an encryption keypair from the account's private key for decryption and generate a random ephemeral keypair for encryption.   
+ 
+Parity wallet already implements a compatible [encrypt/decrypt]  https://wiki.parity.io/JSONRPC-parity-module#parity_decryptmessage method and the MetaMask version is on the way. Having a cross-client standard will enable a whole new wave of decentralized applications that will allow users to securely store their private data in public databases such as IPFS.
+
+### Motivation
+Imagine an illegal immigrant named Martha. Martha moved to the United States illegally but then had 2 children there, so her children are citizens. One day Martha gets arrested and deported but her children get to stay. How will Martha pass power of Attorney, bank account info, identification docs, and other sensitive information to her children? Storing that data in a centralized database can be incriminating for Martha, so maybe decentralized databases like IPFS could help, but if the data is not encrypted anyone can see it, which kind of defeats the purpose. If Martha had access to a Dapp with end-to-end encryption connected to her identity, she could save her data in a decentralized, censor-proof database and still have confidence that only her children can access it.
+
+More casually, Martha can create a treasure hunt game, or a decentralized chat app etc.
+
+### Specification
+ 
+```
+
+const nacl = require('tweetnacl')
+
+/**
```

Comment:
tl;dr=IMHO this is implementation specific, but is worth including here.

An implementation of `web3.eth.getEncryptionPublicKey` would require a call to `nacl.box.keyPair.fromSecretKey(/* ... */)` (or some other equivalent library). `tweetnacl` is a JS implementation of the library referred to in the abstract:  "Nacl is a cryptographically complete and well audited library that works well for this by implementers are free to choose their crypto."

An alternative would be to bundle (parts of) the `tweetnacl` library (or some other equivalent library) together with web3.js. Probably not appropriate for now, but is something to consider if encryption/ decryption use cases (and the use of the JSON-RPC methods introduced in this EIP) become very common in client implementations.

In the case that implementation specific lines do not belong in this section, here's a patch that removes it:

```suggestion
/**
```

FWIW, I think that leaving it in there is probably a good thing because it aids in understanding the proceeding sections of the specifications, and is a reference to something mentioned in the abstract.

addresses: https://github.com/ethereum/EIPs/pull/1098/files#r299563507